nestkernel/recording_device.cpp

```diff
@@ -290,6 +295,14 @@ nest::RecordingDevice::State_::get( DictionaryDatum& d,
       dict, names::weights, std::vector< double >( event_weights_ ) );
   }
 
+  if ( p.withreceivergid_ )
```

Comment:
There was actually a bug in the code. It was meant that only the `weight_recorder` is using `withreceivergid`. But actually, the user can set this parameter to every recording device (i.e. the `spike_detector`), which could lead to undefined behavior as `get_receiver_gid()` was only implemented in the `WeightRecorderEvent`.  I fixed that in 72c7fae, thanks.